src/history/hs_rec.c

```diff
@@ -60,6 +60,57 @@ __hs_verbose_cache_stats(WT_SESSION_IMPL *session, WT_BTREE *btree)
         cache->hs_verb_gen_write = ckpt_gen_current;
 }
 
+/*
+ * __hs_delete_record --
+ *     Delete the value left in the history store
+ */
+static int
+__hs_delete_record(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, WT_ITEM *key,
+  WT_UPDATE *delete_upd, WT_UPDATE *delete_tombstone)
+{
+    WT_DECL_RET;
+    bool hs_read_committed;
+
+#ifdef HAVE_DIAGNOSTIC
+    WT_TIME_WINDOW *hs_tw;
+#endif
+
+    hs_read_committed = F_ISSET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    hs_cursor->set_key(hs_cursor, 4, S2BT(session)->id, key, WT_TS_MAX, UINT64_MAX);
+    WT_ERR_NOTFOUND_OK(__wt_curhs_search_near_before(session, hs_cursor), true);
+    /* It's possible the value in the history store is already obsolete. */
+    if (ret == WT_NOTFOUND) {
+        ret = 0;
```

Comment:
Can you add an assert here to check that this history store entry has its own tombstone? Otherwise, the entry shouldn't be removed.

Reply:
I can't as it returns not found so the cursor is reset.

Reply:
I mean adding an assert to check that `delete_tombstone != NULL`

Reply:
Done